tests/tests/authentication.test.ts

```diff
@@ -0,0 +1,56 @@
+import { authenticationGet, credentialUrl } from './authentication'
+import { TestState } from './test_state'
+import { authorize } from './authorize'
+import { challenge } from './challenge'
+import axios from 'axios'
+import { deriveEncryptionKeyFromSeed } from './utils'
+import * as Kilt from '@kiltprotocol/sdk-js'
+import { toHex } from '@smithy/util-hex-encoding'
+import { CREDENTIAL, DID_KEY_AGREEMENT_URL } from '../test_config'
+import { describe, it, expect } from 'vitest'
+import 'dotenv/config'
+import { EncryptedMessage } from './types'
+
+const mnemonic = process.env.SEED as string
+
+describe('Authentication', () => {
+  it('should refuse unsigned presentation', async () => {
+    const testState = new TestState()
+
+    await authorize(testState)
+    await challenge(testState)
+    await authenticationGet(testState)
+
+    // Use the credential directly without creating a presentation.
+    const payload = {
+      body: { content: [CREDENTIAL], type: 'credential' },
+      createdAt: 0,
+      sender: null,
+      receiver: null,
+      messageId: 0,
+    }
+
+    const seed = Kilt.Utils.Crypto.mnemonicToMiniSecret(mnemonic)
+    const keyAgreementKey = deriveEncryptionKeyFromSeed(seed)
+    const credentialString = JSON.stringify(payload)
+    const encrypted = Kilt.Utils.Crypto.encryptAsymmetric(
+      credentialString,
+      testState.getOpenDidKeyAgreement().publicKey,
+      keyAgreementKey.secretKey,
+    )
```

Comment:
Some of these steps could get done in the setup stage. Not completely necessary but to get it cleaned up.

Reply:
Same here, the tests must run in one stateful flow.

Reply:
Ok, that is fine.